packages/experimental/Icon/src/Icon.types.ts

```diff
@@ -0,0 +1,33 @@
+import { SvgProps } from 'react-native-svg';
+import { AccessibilityProps } from 'react-native';
+export const iconName = 'Icon';
+
+export interface IFontIconProps
+{
+  fontFamily?: string;
+  fontSrcFile?: string;
+  codepoint: number;
+  fontSize?: number;
+}
+
+export interface ISvgIconProps
+{
+  uri?: string;
+  src?: React.FC<SvgProps>;
+  viewBox?: string;
+}
+
+export interface IRasterImageIconProps
+{
+  src: object;
+}
+
+export interface IIconProps extends AccessibilityProps
```

Comment:
This interface should have a style prop as well. You should be able to do something like:

```ts
style?: StyleProp<ImageStyle | TextStyle>
```

then the style should be passed along to the text or image component as appropriate.